style.md

```diff
@@ -0,0 +1,672 @@
+// generally are not allowed.
+myArray.sort((a: number, b: number) => {
+    return a - b;
+});
+```
+
+Do not name loop variables `x` or `i`; use descriptive names for them instead. Prefer identifiers like `vectorIndex` over the less descriptive `index`.
+
+``` ts
+for(let vectorIndex: number = 0; vectorIndex < totalVectors; vectorIndex++){
+    const myVector: Vector = vectors[vectorIndex];
+    doStuffWithVector(myVector);
+}
+```
+
+## Use PascalCase for classes and constants
+
+Class names, constructors, and the names of constants should be written in PascalCase.
+Note that constant in this case does not mean everything declared using `const`, but rather it refers to any variable that is initialized once and never changed during program execution. (Constants are not a matter of syntax in TypeScript, but a matter of intent.)
+
+``` ts
+// [Comment explaining the purpose of my constant]
+const PlaypalLumpName = "PLAYPAL";
+```
+
+``` ts
+// [Comment explaining the purpose of my class]
+class MyClass {
+    // [Comment explaining the purpose of my attribute]
+    myAttribute: string;
+}
+```
+
+## Use camelCase for functions and variables
+
+Names of functions, methods, variables, attributes, and function parameters should be written in camelCase.
+
+``` ts
+// [Comment explaining the usage of this function]
+function getPaletteCount(): number {
+    return 100;
+}
+```
+
+``` ts
+// [Comment explaining the purpose of my class]
+class MyClass {
+    // [Comment explaining the purpose of my attribute]
+    myAttribute: string;
+    
+    // [Comment explaining why this class has a number-squaring helper]
+    mySquaringMethod(value: number): number {
+        return value * value;
+    }
+}
+```
+
+## Use camelCase for TypeScript source file names
+
+TypeScript source files should be assigned pascalCase names.
+
+``` text
+- file.ts
+- fileList.ts
+- fileType.ts
+```
+
+
+# Conventions
+
+## Do not use "eval" or the function constructor
+
+Do not ever use the `eval` built-in function. Do not ever use the `Function` constructor to create a new function. These are inefficient and would make jsdoom vulnerable to arbitrary code execution.
+
+## Do not use "var"
+
+Do not declare variables using `var`. Use `let` or `const` instead.
+
+## Prefer "if" and "else if" over "switch" and "case"
+
+In general, a series of `if` and `else if` statements should be used instead of a `switch` statement.
+
+``` ts
+if(myVariable === 0){
+    doStuff();
+}else if(myVariable === 1){
+    doOtherStuff();
+}else if(myVariable === 2){
+    doYetMoreStuff();
+}else{
+    doDefaultStuff();
+}
+```
+
+## Do not nest ternary expressions
+
+In general, ternary expressions `a ? b : c` should not be nested.
+
+## Prefer strict equality over regular equality
+
+The equality operator `==` and inequality operator `!=` should not generally be used. Prefer the strict equality operator `===` and strict inequality operator `!==` instead.
```

Comment:
TSLint can help enforce some of these rules.

Reply:
Also, double-equals is still often used to compare against null and undefined.

Reply:
Yep, the next code style related step is definitely to add a linter.

Is there a good justification to use `==` instead of `===` when comparing against null and undefined? I have always been partial to using an `isNil` helper if it's necessary to check for both.

Reply:
Strict equality/inequality guideline remains in the form of `6.8.`. Please create an issue or a PR if you still think that this guideline needs to be changed or removed.